src/Message/LearningMaterialTextExtractionRequest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Message;
+
+use InvalidArgumentException;
+
+class LearningMaterialTextExtractionRequest
+{
+    private array $learningMaterialIds;
+    public const int MAX_MATERIALS = 50;
+
+    public function __construct(array $learningMaterialIds)
+    {
+        $count = count($learningMaterialIds);
```

Comment:
this error condition would only be triggered due to a developer error, from what i can see there is no user interaction that would cause this to happen.

suggesting the use of [`assert()`](https://www.php.net/manual/en/function.assert.php) instead of raising an exception here.

Reply:
Ooh, awesome. I've never used this. For now this followed an identical pattern in the other Request handlers. How about if I followup and replace them all after this is merged?

Reply:
sounds good.